CMakeLists.txt

```diff
@@ -130,9 +131,44 @@ IF (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
 
 ENDIF ()
 
-### C++ 20 support
+### C++ 17/20 support
+
+IF (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+
+    IF (BUILD_WITH_CXX_17)
+
+        SET (CMAKE_CXX_STANDARD 17)
+
+        MESSAGE (STATUS "C++17 support enabled")
+
+        IF (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 7.0)
+
+            MESSAGE (FATAL_ERROR "GCC version must be at least 7.0")
+
+        ENDIF ()
+
+    ELSE ()
+
+        MESSAGE (STATUS "C++20 support enabled")
+
+        SET (CMAKE_CXX_STANDARD 20)
+
+        IF (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 13.0)
+
+            MESSAGE (FATAL_ERROR "GCC version must be at least 13.0")
+
+        ENDIF ()
+
+    ENDIF ()
+
+    SET (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -Wpedantic -Wshadow -Wno-deprecated")
+
+ELSE ()
+
+    SET (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -Wpedantic -Wshadow")
+
+ENDIF ()
 
-SET (CMAKE_CXX_STANDARD 20)
 SET (CMAKE_CXX_STANDARD_REQUIRED ON)
 SET (CMAKE_CXX_EXTENSIONS OFF)
```

Comment:
Do you not need the Dockerfile changes here to add the format lib dependency for the sub-libraries?

Reply:
It was never deleted in OSTk simulation. I had missed it last time.